fix: pfe-autocomplete update aria-selected attribute

kelsS opened the pull request:
## <Pull request description>

Add feature to update aria-selected attribute to add the `aria-selected="true"` to the currently active list item when keying up and down through the droplist list items.


### Related issues

- n/a


### Preview

<!-- Suggest linking to Netlify or a public sandbox; not a resource behind a log-in or VPN -->
Link(s) to demo page(s) where this element can be viewed:
- [Autocomplete Demo](https://deploy-preview-1713--patternfly-elements.netlify.app/elements/pfe-autocomplete/demo/) 

### Testing instructions

<!-- Be sure to include detailed instructions on how your update can be tested by another developer. -->

  1. Go to the [demo page](https://deploy-preview-1713--patternfly-elements.netlify.app/elements/pfe-autocomplete/demo/)
  2. Click in side of the autocomplete input
  3. Inspect the page and find the `.droplist ul li` items
  4. Starting type the letter `r` in the input
  5. Press the `arrow` down key and double check that the `aria-selected="true"` is added and removed as you key through the list
  6. The attribute should exist on the currently active list item only

#### Browser requirements

Your component should work in all of the following environments:

- [ ] Latest 2 versions of Edge
- [ ] Internet Explorer 11 (should be useable, not pixel perfect)
- [ ] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [ ] Firefox 78 (or latest version for Red Hat Enterprise Linux distribution)
- [ ] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [ ] Latest 2 versions of Safari
- [ ] Android mobile device (such as the Galaxy S9)
- [ ] Apple mobile device (such as the iPhone X)
- [ ] Apple tablet device (such as the iPhone Pro)


### Ready-for-merge Checklist

<!-- Check off items as they are completed.  Feel free to delete items if they are not applicable. -->

- [ ] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [ ] Tests have been updated to cover these changes.
- [ ] Browser testing passed.
- [ ] Changelog updated.
- [ ] Documentation (README.md, WHY.md, etc.) updated or added.
- [ ] Link to the demo recording: []()
- [ ] Approved by designer.


### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

castastrophe commented on elements/pfe-autocomplete/src/pfe-autocomplete.js:
```diff
@@ -309,14 +309,23 @@ class PfeAutocomplete extends PFElement {
 
+  /**
+   * Returns the HTML of the active element
+   * @param {integer} activeIndex Index of an element in the droplist
```
```suggestion
   * @param {number} activeIndex Index of an element in the droplist
```

Integer is not a valid JSDoc type so I would suggest using `number` here instead.

kelsS replied:
@castastrophe thanks for catching that! I will get that updated.

castastrophe commented on elements/pfe-autocomplete/src/pfe-autocomplete.js:
```diff
@@ -309,14 +309,23 @@ class PfeAutocomplete extends PFElement {
+  /**
+   * Returns the HTML of the active element
+   * @param {integer} activeIndex Index of an element in the droplist
+   * @return The HTML inside of the given index
```
```suggestion
   * @return {string} The HTML inside of the given index as a string
```

I updated the return note here but (and I know this is out of scope of the work you're doing), but I might suggest returning the element itself for microfunctions like this in the future - it's more reusable.

kelsS replied:
@castastrophe ahh gotcha, that makes sense. I really appreciate the feedback!

castastrophe replied:
Thanks for adding all this awesome documentation 🎉

castastrophe commented on elements/pfe-autocomplete/src/pfe-autocomplete.js:
```diff
@@ -309,14 +309,23 @@ class PfeAutocomplete extends PFElement {
     return this._dropdown.shadowRoot.querySelector("li:nth-child(" + (parseInt(activeIndex, 10) + 1) + ")").innerHTML;
   }
 
+  /**
+   * Handle keyboard input, we care about arrow keys, enter, and escape
+   */
```
```suggestion
  /**
   * Handle keyboard input, we care about arrow keys, enter, and escape
   * @param {object} e - keypress event 
   */
```

castastrophe commented on elements/pfe-autocomplete/src/pfe-search-droplist.scss:
```diff
@@ -12,6 +12,10 @@
 }
 
+:host([debug]) [aria-selected] {
+  box-shadow: inset 0 0 0 2px red;
```
```suggestion
  box-shadow: inset 0 0 0 pfe-var(ui--border-width--md) pfe-var(feedback--critical);
```

If you still need this code in the production version, can you update it to use the variables?  If not, feel free to delete.

kelsS replied:
@castastrophe good call. I am going to go ahead and keep the styles since it is really helpful for debugging. I will get the color updated to the appropriate pfe-vars. 🙂

kylebuch8 reviewed (approved):
Leaves Germany: Tschüss, Mama!

kylebuch8 commented:
Oh @kelsS! I forgot to ask for an update to the tests. Can you do that?

kelsS commented:
Hi @kylebuch8 I updated the tests and added some aria attributes that were missing. I also fixed some a11y errors on the demo page. The component now has a 100 score on Lighthouse. I believe it previously was an 88. I will run a Lighthouse test on the original before we merge this so that we can see the progress.

[pfe-autocomplete-lh-report-pr-#1713.pdf](https://github.com/patternfly/patternfly-elements/files/6881730/pfe-autocomplete-lh-report-pr-.1713.pdf)
